MusicApp.PlaylistService.Infrastructure/Repositories/PlaylistRepository.cs

```diff
@@ -33,9 +37,22 @@ public async Task<IEnumerable<Playlist>> GetMyPrivatePlaylistsAsync(string usern
 
     public override async Task<Playlist?> GetByIdAsync(Guid id, CancellationToken cancellationToken)
     {
-        return await _appContext.Playlists
+        var playlist = await _cache.GetEntityAsync<Playlist>(id.ToString(), cancellationToken);
+        if (playlist != null)
+        {
+            return playlist;
+        }
+
+        playlist = await _appContext.Playlists
             .Include(playlist => playlist.Songs)
             .Include(playlist => playlist.Creator)
             .FirstOrDefaultAsync(playlist => playlist.Id == id, cancellationToken);
+
+        if (playlist != null)
+        {
+            await _cache.SetEntityAsync(id.ToString(), playlist, cancellationToken);
```

Comment:
it's okay, but you can consider about defining something like GetOrSetEntity method in DistributedCacheExtension to extract this logic from each repository to single place to avoid code duplication

Reply:
But can I inject the db context into the DistributedCacheExtension?

Reply:
you have to modify your implementation of DistributedCacheExtension and define something like RedisCacheService instead of it and inject IConnectionMultiplexer there, and then use this service in your repositories, but it's not necessary, you can consider about it later if you will have time for this